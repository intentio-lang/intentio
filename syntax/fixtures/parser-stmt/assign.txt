e = r#"my sweet "string""#;
e = e + "2"
