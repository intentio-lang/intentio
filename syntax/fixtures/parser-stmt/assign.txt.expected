[
  {
    "_stmtAnn": [],
    "_stmtKind": {
      "contents": [
        {
          "_scopeIdAnn": [],
          "_scopeIdSourcePos": [
            "test.ieo",
            0,
            0
          ],
          "_unScopeId": "e"
        },
        {
          "_exprAnn": [],
          "_exprKind": {
            "contents": {
              "_litAnn": [],
              "_litKind": {
                "contents": "r#\"my sweet \"string\"\"#",
                "tag": "RawStringLit"
              },
              "_litSourcePos": [
                "test.ieo",
                0,
                4
              ]
            },
            "tag": "LitExpr"
          },
          "_exprSourcePos": [
            "test.ieo",
            0,
            4
          ]
        }
      ],
      "tag": "AssignStmt"
    },
    "_stmtSourcePos": [
      "test.ieo",
      0,
      0
    ]
  },
  {
    "_stmtAnn": [],
    "_stmtKind": {
      "contents": [
        {
          "_scopeIdAnn": [],
          "_scopeIdSourcePos": [
            "test.ieo",
            1,
            0
          ],
          "_unScopeId": "e"
        },
        {
          "_exprAnn": [],
          "_exprKind": {
            "contents": [
              {
                "_binOpAnn": [],
                "_binOpKind": "BinAdd",
                "_binOpSourcePos": [
                  "test.ieo",
                  1,
                  6
                ]
              },
              {
                "_exprAnn": [],
                "_exprKind": {
                  "contents": {
                    "contents": {
                      "_scopeIdAnn": [],
                      "_scopeIdSourcePos": [
                        "test.ieo",
                        1,
                        4
                      ],
                      "_unScopeId": "e"
                    },
                    "tag": "ScopeId'"
                  },
                  "tag": "IdExpr"
                },
                "_exprSourcePos": [
                  "test.ieo",
                  1,
                  4
                ]
              },
              {
                "_exprAnn": [],
                "_exprKind": {
                  "contents": {
                    "_litAnn": [],
                    "_litKind": {
                      "contents": "\"2\"",
                      "tag": "StringLit"
                    },
                    "_litSourcePos": [
                      "test.ieo",
                      1,
                      8
                    ]
                  },
                  "tag": "LitExpr"
                },
                "_exprSourcePos": [
                  "test.ieo",
                  1,
                  8
                ]
              }
            ],
            "tag": "BinExpr"
          },
          "_exprSourcePos": [
            "",
            0,
            0
          ]
        }
      ],
      "tag": "AssignStmt"
    },
    "_stmtSourcePos": [
      "test.ieo",
      1,
      0
    ]
  }
]
